### PR TITLE
Decorate class

### PR DESCRIPTION
**Pull request by OsvaldoBC**
# Good day! 🌤️ 

The milestone in this pull request is :
to use the Decorator design pattern to correct the names of people.

## 🛠️ Changes made
✔️ Create a Nameable class.
✔️ Modify the class called Person to inherit from Nameable.
✔️ Create the class called  Decorator from Nameable.
✔️ Create the class called CapitalizeDecorator and TrimmerDecorator from Decorator.
✔️ Test correct_name method functionality.

I'm expecting feedback.

## Happy coding! 👏👏👏

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require './nameable'
+
+# class decorator
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
```
Comment (VICTOR-CHUKWU) [inline decorator.rb:6]: - [ ] nameable should be made accessible i.e ` attr_accessor :nameable`

File: decorator.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    super()
+  end
+
+  def correct_name
+    super()
+    @nameable
+  end
```
Comment (VICTOR-CHUKWU) [inline decorator.rb]: - [ ] this method should return the result of the correct_name method of the @namable i.e 
```
 def correct_name
      @nameable.correct_name
    end
```

File: capitalize_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require './decorator'
+
+# class capitalize_decorator
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.capitalize
+  end
```
Comment (VICTOR-CHUKWU) [inline capitalize_decorator.rb]: - [ ] this method should also return a capitalized correct_name method form of nameable
```
 def correct_name
      @nameable.correct_name.capitalize
    end
```

File: trimmer_decorator.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+require './decorator'
+require './capitalize_decorator'
+require './person'
+
+# class trimmer_decorator
+class TrimmerDecorator < Decorator
+  def correct_name
+    super()
+    @nameable.length > 10 ? @nameable[0..9] : @nameable
```
Comment (VICTOR-CHUKWU) [inline trimmer_decorator.rb]: - [ ] also in this  should return a trimed correct_name method of nameable `@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

File: nameable.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# class nameable
+class Nameable
+  def correct_name
+    raise 'NotImplementedError'
```
Comment (VICTOR-CHUKWU) [inline nameable.rb:4]: - [ ] [Optional] the error message can be made more specific `raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"`

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @OsvaldoBC 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @OsvaldoBC, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**